### ISSUE TITLE
Move bundle hosting to S3

### ISSUE THREAD

**Issue by timolegros** [hicommonwealth/commonwealth#8017]
## Description
We should move the bundle hosting/serving to S3 so that the Heroku app can focus on handling API requests. This will remove the performance impact that sending the bundle to the client has on the web dynos.

## Additional context

How will this affect prerendering (prerender.io)?

**Comment by kurtisassad** [comment]
Some thoughts after digging into this a bit.

## Local development will need to be changed. 
(Suggestion: should be completed separately before the frontend infra migration)

We can only route our apex domain (commonwealth.im) to a single target. But since our front end will be served from s3, and our backend served separately, in this case we would need to reconfigure our routing, becuase the same domain can no longer serve both front and backend.

I think in this case we could CNAME api.commonwealth.im to the backend, and make all api calls to api.commonwealth.im. In this case the vite dev server will need to proxy /api calls to our backend but remove the /api portion in development mode. And then in production mode we would just directly route to the api.commonwealth.im

## We should migrate over to commonwealth.xyz

This will become apparent in the following sections, but basically we should have commonwealth.xyz point to our s3 bucket, and api.commonewalth.xyz point to our heroku backend. This would also require migrating our assets and sitemap buckets to the new commonewalth.xyz domain.

## Custom domains will need to be re-distributed

When switching front end providers from Heroku to s3, we will need to regenerate DNS targets for all our custom domains and re-distribute them to the customers. This will be challenging and I have not thought up of a proper solution of a way to do this gradually. For example lets take https://gov.1inch.io/. If we regenerate the DNS target for s3 and distribute it to 1inch, they will need to change their DNS target at gov.1inch.io to now point to the s3 bucket. The issue with this is that we will need to ensure we have both front ends serving (Both on s3 and heroku) in order to ensure that there is a smooth transition without any downtime on the customers side. This of course conflicts with the fact that we can only point our apex domain to one dns target. This means that we cannot have both running at the same time. This is why I had suggested that it would be a good time to gradually migrate off to commonwealth.xyz. In this scenario we would be able to gradually migrate off new customers while still keeping the custom domains valid for the old customers.

## Some solutions/suggestions to questions

> How will this affect prerendering (prerender.io)?

If we take the previous sections suggestions and perform this migration while migrating over to commonwealth.xyz, then both prerender and magic (All I can think of now, maybe more) will need to be reconfigured and whitelisted for our new site at commonwealth.xyz. If we perform this gradual migration it might be possible to prerender with the same prerender.io account, or we could always create a new account for the specific new domain.

> We will need to consolidate all our express.static calls into one bundle call (no more separate robots.txt, favicon, ...).

Luckily this part is already done with the vite migration. Basically we still serve these from express right now, but they are all actually consolidated in the frontend bundle, so they are ready to be detached from the current heroku build process.

To wrap this up, I think this would be a large migration, but it would slowly migrate us off of heroku which we plan to do in the future, as well as improve the our frontend scaling. It would also de-correlate the cpu/memory usage of frontend related tasks with the api server, which would make it easier for us to monitor and analyze our api server (like gzipping the files, or reading them from disk into buffers).

**Comment by timolegros** [comment]
Rather than migrating to `commonwealth.xyz` we should probably just use `common.xyz`. Seems like migrating static content to external storage should be done in tandem to migration to `common.xyz`

**Comment by kurtisassad** [comment]
Yep common.xyz, didn't know we owned and were planning to migrate to this domain
